openwisp-monitoring/files/monitoring.agent

```diff
@@ -182,6 +184,13 @@ send_data() {
 					# remove saved data
 					rm -f "$filename"
 					break
+				elif [ "$response_code" = "400" ]; then
```

Comment:
Shall we do the same for any 4XX response? Any error response in the range of 400-499 is meant to be a client error and hence is always going to be rejected by the server unless the client modifies the request.

Reply:
I have some doubt regarding that. If we are getting client error, does that mean there is something wrong with the data? It can be due to bad config too.

In such cases, fixing the cause can help sending the data collected previously.

Reply:
The other 4XX response type we could get is 404 which I believe will have to be handled as  https://github.com/openwisp/openwisp-config/issues/160, for this I have created https://github.com/openwisp/openwrt-openwisp-monitoring/issues/105, we can leave this code as is for now.